src/Avalonia.Controls/Primitives/PopupRoot.cs

```diff
@@ -75,6 +76,29 @@ public PopupRoot(IAvaloniaDependencyResolver dependencyResolver)
         /// <inheritdoc/>
         public void Dispose() => PlatformImpl?.Dispose();
 
+        /// <summary>
+        /// Moves the Popups position so that it doesnt overlap screen edges.
+        /// This method can be called immediately after Show has been called.
+        /// </summary>
+        public void SnapInsideScreenEdges()
+        {
+            var mainWindow = Window.OpenWindows.First();
```

Comment:
It's weird to me that we have to go through the `Window.OpenWindows` collection to get a `Screens` object in `PopupRoot`. Any chance we can move the `Screens` property on `Window` up to `WindowBase` so `PopupRoot` can have direct access to the Screens object?

cc: @grokys @kekekeks

Reply:
@jkoritzinsky isn't it weird that we need a window at all?

Surely Screens doesn't need to depend on the window class and we could simply have

Screens.Instance always available?

Reply:
But isn't screen a device itself? A screen is independent of a window and doesn't need a window to exist. In the end you always need a screen first. Screens could hold physical and virtual devices. Just my thoughts.

Reply:
The Screens object on the Window doesn’t represent all screens on all devices. It represents an arrangement of a set of screens that form a distinct coordinate system. Each window is associated with one set of screens. In theory we could have an object that is a list of Screens objects for each set of screens (or a list of Screen objects), but it wouldn’t be super useful.

Reply:
@jkoritzinsky I agree that some kind of refactoring needs to take place with screens, but do we need an answer on that before we can approve this?

Reply:
Not really. Any chance we can make a change though that searches up the tree to find the Window the popup is being placed on? That way we can be sure we are using the correct screen edges. That’s the main thing I was worried about here.

Reply:
ok I will take a look at doing that :)

Reply:
@jkoritzinsky I believe it now walks the tree correctly.

I guess this is why screens should be on popup, because this code wouldnt work on platforms where we dont have windows.